skopt/space.py

```diff
@@ -51,7 +61,7 @@ def fit(self, X):
         * `X` [array-like, shape=(n_categories,)]:
             List of categories.
         """
-        self._lb.fit(X)
+        self._lb.fit(encode_none(X))
```

Comment:
I think the use of `np.vectorize` and these extra functions is a bit overkill. Also, I suspect `np.vectorize` may introduce subtle differences due to type coercion (e.g with `otypes="O"`). 

In legacy Python we indeed have:

```
In [1]: import numpy as np

In [2]: s = ["a", "b", None]

In [3]: f = np.vectorize(lambda v: v if v is not None else "__None__", otypes="U")

In [4]: g = np.vectorize(lambda v: v if v != "__None__" else None, otypes="O")

In [5]: f(s)
Out[5]: 
array([u'a', u'b', u'__None__'], 
      dtype='<U8')

In [6]: g(f(s))
Out[6]: array([u'a', u'b', None], dtype=object)

In [7]: g(f(s))[0]
Out[7]: u'a'

In [8]: type(g(f(s))[0])
Out[8]: unicode

In [9]: type(s[0])
Out[9]: str
```

Instead, you could directly iterate over the values of X:

```self._lb.fit([v if v is not None else "__None__" for v in X])```